### DISCUSSION TITLE
I can't install Directus 9 in Microsoft Azure

### DISCUSSION THREAD

**Discussion by ghezeli** [directus/directus#9946] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Hi
I try install Directus 9 in Microsoft Azure  Cloud at https://portal.azure.com/
with docker-compose in https://docs.directus.io/getting-started/installation/docker/ 
but it's not working
I attached screenshot of setting and logs
![1](https://user-images.githubusercontent.com/5620958/142734707-3fb8b13e-5de0-447f-91ff-be2ebc4ee082.png)
![2](https://user-images.githubusercontent.com/5620958/142734709-53eb2542-a449-4dfe-82aa-96cf79c8924a.png)
![3](https://user-images.githubusercontent.com/5620958/142734710-55124c2b-8eeb-433c-a485-09908a73e1a0.png)

### To Reproduce

version: '3'
services:
  database:
    container_name: database
    image: postgis/postgis:13-master
    volumes:
      - ./data/database:/var/lib/postgresql/data
    networks:
      - directus
    environment:
      POSTGRES_USER: 'directus'
      POSTGRES_PASSWORD: 'directus'
      POSTGRES_DB: 'directus'

  cache:
    container_name: cache
    image: redis:6
    networks:
      - directus

  directus:
    container_name: directus
    image: directus/directus:latest
    ports:
      - 8000:8055
    volumes:
      # By default, uploads are stored in /directus/uploads
      # Always make sure your volumes matches the storage root when using
      # local driver
      - ./uploads:/directus/uploads
      # Make sure to also mount the volume when using SQLite
      # - ./database:/directus/database
      # If you want to load extensions from the host
      # - ./extensions:/directus/extensions
    networks:
      - directus
    depends_on:
      - cache
      - database
    environment:
      KEY: '255d861b-5ea1-5996-9aa3-922530ec40b1'
      SECRET: '6116487b-cda1-52c2-b5b5-c8022c45e263'

      DB_CLIENT: 'pg'
      DB_HOST: 'database'
      DB_PORT: '5432'
      DB_DATABASE: 'directus'
      DB_USER: 'directus'
      DB_PASSWORD: 'directus'

      CACHE_ENABLED: 'true'
      CACHE_STORE: 'redis'
      CACHE_REDIS: 'redis://cache:6379'

      ADMIN_EMAIL: 'admin@example.com'
      ADMIN_PASSWORD: 'd1r3ctu5'

      # Make sure to set this in production
      # (see https://docs.directus.io/reference/environment-variables/#general)
      # PUBLIC_URL: 'https://directus.example.com'

networks:
  directus:

### What version of Directus are you using?

9.0.1

### What version of Node.js are you using?

16.12.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

Linux 5.4.0-81-generic

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
Heya! That error seems to suggest there's a syntax error in the YAML file used for the definition, rather than a problem in directus itself. 

We recently launched an azure starter sample in collaboration with the Azure team. That might be a good starting point for inspiration  on how to configure the Azure ARM template 😄

https://github.com/Azure-Samples/azure-sql-db-rest-graphql-directus

**Reply by ghezeli** [reply]
tanks for your replay
how can i install directus with docker compose on Azure?

**Reply by rijkvanzanten** [reply]
I don't have any experience deploying to Azure through Docker Compose. Maybe https://docs.microsoft.com/en-us/azure/container-instances/tutorial-docker-compose combined with https://docs.directus.io/getting-started/installation/docker/#docker-compose could be helpful 👍🏻

**Reply by ghezeli** [reply]
tanks, I try it before, but not working

> version: '3'
services:
  database:
    container_name: database
    image: postgis/postgis:13-master
    environment:
      POSTGRES_USER: 'directus'
      POSTGRES_PASSWORD: 'directus'
      POSTGRES_DB: 'directus'

  cache:
    container_name: cache
    image: redis:6

  directus:
    container_name: directus
    image: directus/directus:latest
    ports:
      - 8000:8055

    depends_on:
      - cache
      - database
    environment:
      KEY: '255d861b-5ea1-5996-9aa3-922530ec40b1'
      SECRET: '6116487b-cda1-52c2-b5b5-c8022c45e263'

      DB_CLIENT: 'pg'
      DB_HOST: 'database'
      DB_PORT: '5432'
      DB_DATABASE: 'directus'
      DB_USER: 'directus'
      DB_PASSWORD: 'directus'

      CACHE_ENABLED: 'true'
      CACHE_STORE: 'redis'
      CACHE_REDIS: 'redis://cache:6379'

      ADMIN_EMAIL: 'admin@example.com'
      ADMIN_PASSWORD: 'd1r3ctu5'

![image](https://user-images.githubusercontent.com/5620958/143008731-7fea5e38-bd0b-4d8a-bb04-d034476829ec.png)

**Reply by azrikahar** [reply]
@ghezeli A search based on the error "Stopping site because it failed during startup" led me to this thread here: https://docs.microsoft.com/en-us/answers/questions/420922/docker-compose-keep-showing-info-39stopping-site-x.html, which should answer your question here. So just to be clear, this should be something to do with Azure App Service, not Directus. 👍 Particularly the fact that it expects the app the listen to port 80 by default, or you can set the `WEBSITE_PORT` app setting to 8055: https://docs.microsoft.com/en-us/azure/app-service/configure-custom-container?pivots=container-linux#configure-port-number

If you would like to switch the port 8055 which Directus listens to by default to port 80, you can use the `PORT` environment variable ([which is app setting for Azure App Service](https://docs.microsoft.com/en-us/azure/app-service/configure-common#configure-app-settings)) instead.

**Reply by ghezeli** [reply]
Tanks a lot for your replay
I can create a directus by this link:
https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure-Samples%2Fazure-sql-db-rest-graphql-directus%2Fmain%2Fazuredeploy.json

but i want to use mysql for database, how can i do this?

**Reply by torstenlemke** [reply]
> but i want to use mysql for database, how can i do this?

In fact I'm not 100% sure _where you'd need to configure these values in azure_ but you will need a (external) MySQL Server which needs to be configure via environment params 

e.g. postgres 
```
  DB_CLIENT: 'pg'
  DB_HOST: 'database'
  DB_PORT: '5432'
  DB_DATABASE: 'directus'
  DB_USER: 'directus'
  DB_PASSWORD: 'directus'

```
_this is perfectly described in the docs_

If you need a build-in MySQL server you would need to edit the whole creation process and connect via environment to that local MySQL, which is (for the most cases) not a good idea in terms of db hosting.

**Reply by rijkvanzanten** [reply]
> Tanks a lot for your replay I can create a directus by this link: https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure-Samples%2Fazure-sql-db-rest-graphql-directus%2Fmain%2Fazuredeploy.json
> 
> but i want to use mysql for database, how can i do this?

That template includes a MS SQL database, so it won't automatically allow you to substitute that for a MySQL database. I would recommend duplicating that template, and adjusting it to fit your exact needs 👍🏻
